hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveSchemaUtil.java

```diff
@@ -155,113 +150,75 @@ private static boolean isFieldExistsInSchema(Map<String, String> newTableSchema,
-      GroupType parquetGroupType = parquetType.asGroupType();
-      OriginalType originalType = parquetGroupType.getOriginalType();
-      if (originalType != null) {
-        switch (originalType) {
-          case LIST:
-            if (parquetGroupType.getFieldCount() != 1) {
-              throw new UnsupportedOperationException("Invalid list type " + parquetGroupType);
-            }
-            Type elementType = parquetGroupType.getType(0);
-            if (!elementType.isRepetition(Type.Repetition.REPEATED)) {
-              throw new UnsupportedOperationException("Invalid list type " + parquetGroupType);
-            }
-            return createHiveArray(elementType, parquetGroupType.getName());
-          case MAP:
-            if (parquetGroupType.getFieldCount() != 1 || parquetGroupType.getType(0).isPrimitive()) {
-              throw new UnsupportedOperationException("Invalid map type " + parquetGroupType);
-            }
-            GroupType mapKeyValType = parquetGroupType.getType(0).asGroupType();
-            if (!mapKeyValType.isRepetition(Type.Repetition.REPEATED)
-                || !mapKeyValType.getOriginalType().equals(OriginalType.MAP_KEY_VALUE)
-                || mapKeyValType.getFieldCount() != 2) {
-              throw new UnsupportedOperationException("Invalid map type " + parquetGroupType);
-            }
-            Type keyType = mapKeyValType.getType(0);
-            if (!keyType.isPrimitive()
-                || !keyType.asPrimitiveType().getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.BINARY)
-                || !keyType.getOriginalType().equals(OriginalType.UTF8)) {
-              throw new UnsupportedOperationException("Map key type must be binary (UTF8): " + keyType);
-            }
-            Type valueType = mapKeyValType.getType(1);
-            return createHiveMap(convertField(keyType), convertField(valueType));
-          case ENUM:
-          case UTF8:
-            return "string";
-          case MAP_KEY_VALUE:
-            // MAP_KEY_VALUE was supposed to be used to annotate key and
-            // value group levels in a
-            // MAP. However, that is always implied by the structure of
-            // MAP. Hence, PARQUET-113
-            // dropped the requirement for having MAP_KEY_VALUE.
-          default:
-            throw new UnsupportedOperationException("Cannot convert Parquet type " + parquetType);
+        noNullSchemaFields.add(new ImmutablePair<String, Schema>(fieldItem.name(), fieldItem.schema()));
+      }
+      return createHiveStructFromAvro(noNullSchemaFields);
+    } else if (type.equals(Schema.Type.ENUM)) {
+      return field.append("string").toString();
+    } else if (type.equals(Schema.Type.ARRAY)) {
+      Schema elementType = schema.getElementType();
+      String schemaName = schema.getName();
+      return createHiveArrayFromAvro(schemaName, elementType);
+    } else if (type.equals(Schema.Type.MAP)) {
+      String keyType = "string";
```

Comment:
Why fix the key type for Map datatype as strings, instead of converting the original key type ?